python/lsst/multiprofit/sourceconfig.py

```diff
@@ -0,0 +1,385 @@
+                    is_final=is_final,
+                )
+                # TODO: Omitting this crucial step should raise but doesn't
+                # There shouldn't be two IntegralModels with the same last
+                # especially not one is_final and one not
+                last = integral_model
+            else:
+                integral_model = config_comp.make_linear_integral_model(
+                    fluxes_component,
+                    label_integral=label_integral_comp,
+                )
+
+            centroid = self.centroids.get(name_component)
+            if not centroid:
+                if centroid_default is None:
+                    centroid_default = self.centroids["default"].make_centroid()
+                centroid = centroid_default
+            componentdata = config_comp.make_component(
+                centroid=centroid,
+                integral_model=integral_model,
+            )
+            components.append(componentdata.component)
+            priors.extend(componentdata.priors)
+        return components, priors
+
+    def validate(self):
+        super().validate()
+        errors = []
+        components: ComponentConfigs = dict(self.components_gauss)
+
+        for name, component in self.components_sersic.items():
+            if name in components:
+                errors.append(
+                    f"key={name} cannot be used in both self.components_gauss and self.components_sersic"
+                )
+            components[name] = component
+
+        keys = set(self.centroids.keys())
+        has_default = "default" in keys
+        for name in components.keys():
+            if name in keys:
+                keys.remove(name)
+            elif not has_default:
+                errors.append(f"component {name=} has no entry in self.centroids and default not specified")
+        if errors:
+            newline = "\n"
+            raise ValueError(f"ComponentMixtureConfig has validation errors:\n{newline.join(errors)}")
+
+
+class SourceConfig(pexConfig.Config):
+    """Configuration for a gauss2d.fit Source.
+
+    Sources may contain components with distinct centroids that may be linked
+    by a prior (e.g. a galaxy + AGN + star clusters),
+    although such priors are not yet implemented.
+    """
+
+    component_groups = pexConfig.ConfigDictField[str, ComponentGroupConfig](
+        doc="Components in the source",
+        optional=False,
+    )
+
+    def _make_components_priors(
+        self,
+        component_group_fluxes: list[list[Fluxes]],
+        label_integral: str,
+        validate_psf: bool = False,
+    ) -> [list[g2f.Component], list[g2f.Prior]]:
+        if len(component_group_fluxes) != len(self.component_groups):
+            raise ValueError(f"{len(component_group_fluxes)=} != {len(self.component_groups)=}")
+        components = []
+        priors = []
+        if validate_psf:
+            keys_expected = tuple((g2f.Channel.NONE,))
```

Comment:
You shouldn't need to wrap with `tuple` here. Also, this seems incomplete. Shouldn't this be a tuple of channels, or is there always only a single null channel? A comment to explain in the code would be nice.

Reply:
It's intentional; PSFs must only use the NONE channel (so that they can be a generic model*, not attached to a particular Observation). I added a note to this effect.

*There isn't as yet an obvious benefit to this design choice, but I suppose we'll see. It does mean you can use the same PSF model for multiple observations with different channels, although I'm not sure why anyone would want to.